### PR TITLE
fix: Add check for settlementType of Aggregated documentWriters and versioning of internal commands

### PR DESCRIPTION
**Pull request by RasmusGraabaek**
## Description
This PR will fix the error raised in this [link](https://portal.azure.com/#view/AppInsightsExtension/DetailsV2Blade/ComponentId~/%7B%22SubscriptionId%22%3A%226383f566-3f4f-4690-86ee-c5a4382b29b3%22%2C%22ResourceGroup%22%3A%22rg-shres-d-we-001%22%2C%22Name%22%3A%22appi-shres-d-we-001%22%2C%22LinkedApplicationType%22%3A0%2C%22ResourceId%22%3A%22%252Fsubscriptions%252F6383f566-3f4f-4690-86ee-c5a4382b29b3%252FresourceGroups%252Frg-shres-d-we-001%252Fproviders%252FMicrosoft.Insights%252Fcomponents%252Fappi-shres-d-we-001%22%2C%22ResourceType%22%3A%22microsoft.insights%252Fcomponents%22%2C%22IsAzureFirst%22%3Afalse%7D/DataModel~/%7B%22eventId%22%3A%22ee383516-e508-11ee-b3a1-000d3a4a88db%22%2C%22timestamp%22%3A%222024-03-18T09%3A20%3A49.492Z%22%2C%22cacheId%22%3A%227894a157-89d3-4673-a149-cfba7903ad7f%22%2C%22eventTable%22%3A%22exceptions%22%7D)

The fix takes the settlement type from the wholesale response instead of our process.
With the current implementation, it's impossible to send a flex-, nonprofiled- comsumption message, since these will always be mapped to total-comsumption. Which is an error.

Furthermore enqueued internal commands has a version

Has been [deployed](https://github.com/Energinet-DataHub/dh3-environments/actions/runs/8344857686/job/22838676018)

### CODE REVIEW

File: source/Process.Application/Transactions/Aggregations/EnergyResultTimeSeriesRequestAcceptedEventMapper.cs [hunk 1]
```diff
@@ -74,16 +75,31 @@ public bool CanHandle(string eventType)
...
+    {
+        return timeSeriesType switch
+        {
+            TimeSeriesType.Production => null,
+            TimeSeriesType.FlexConsumption => SettlementType.Flex,
+            TimeSeriesType.NonProfiledConsumption => SettlementType.NonProfiled,
+            TimeSeriesType.NetExchangePerGa => null,
+            TimeSeriesType.NetExchangePerNeighboringGa => null,
+            TimeSeriesType.TotalConsumption => null,
             TimeSeriesType.Unspecified => throw new InvalidOperationException("Unknown metering point type"),
```
Comment (lasrinnil) [inline source/Process.Application/Transactions/Aggregations/EnergyResultTimeSeriesRequestAcceptedEventMapper.cs:103]:
```suggestion
            TimeSeriesType.Unspecified => throw new InvalidOperationException("Unknown metering point type for determining settementtype"),
```

File: source/Process.Application/Transactions/Aggregations/EnergyResultTimeSeriesRequestAcceptedEventMapper.cs [hunk 2]
```diff
@@ -74,16 +75,31 @@ public bool CanHandle(string eventType)
...
             _ => throw new InvalidOperationException("Could not determine metering point type"),
```
Comment (lasrinnil) [inline source/Process.Application/Transactions/Aggregations/EnergyResultTimeSeriesRequestAcceptedEventMapper.cs:104]:
```suggestion
            _ => throw new InvalidOperationException("Could not determine settlementtype from metering point type),
```

File: source/Process.Domain/Transactions/AggregatedMeasureData/AcceptedEnergyResultTimeSerie.cs [hunk 3]
```diff
@@ -22,9 +22,10 @@ namespace Energinet.DataHub.EDI.Process.Domain.Transactions.AggregatedMeasureDat
 [Serializable]
 public record AcceptedEnergyResultTimeSerie(
     IReadOnlyCollection<Point> Points,
-    string MeteringPointType,
-    string UnitType,
-    string Resolution,
+    MeteringPointType MeteringPointType,
```
Comment (lasrinnil) [inline source/Process.Domain/Transactions/AggregatedMeasureData/AcceptedEnergyResultTimeSerie.cs:25]: 🙏

File: source/IntegrationTests/Application/Transactions/WholesaleServices/WhenAnAcceptedWholesaleServicesResultIsAvailableTests.cs [hunk 4]
```diff
@@ -143,6 +147,44 @@ public async Task Received_2_accepted_wholesale_services_events_enqueues_2_messa
         outgoingMessages.Count.Should().Be(2);
     }
 
+    [Fact]
+    public async Task Received_same_accepted_wholesale_services_event_can_write_document_in_json()
```
Comment (lasrinnil) [inline source/IntegrationTests/Application/Transactions/WholesaleServices/WhenAnAcceptedWholesaleServicesResultIsAvailableTests.cs:151]: I dont get the intent of this test?

File: source/Process.Application/Extensions/DependencyInjection/InternalCommandExtensions.cs [hunk 5]
```diff
@@ -43,13 +43,14 @@ internal static IServiceCollection AddInternalCommands(this IServiceCollection s
...
-        mapper.Add("SendAggregatedMeasureDataRequestToWholesale", typeof(SendAggregatedMeasureDataRequestToWholesale));
-        mapper.Add("AcceptedAggregatedTimeSerie", typeof(AcceptedEnergyResultTimeSerieCommand));
-        mapper.Add("RejectedAggregatedTimeSeries", typeof(RejectedAggregatedTimeSeries));
-        mapper.Add("SendWholesaleServicesRequestToWholesale", typeof(SendWholesaleServicesRequestToWholesale));
-        mapper.Add("RejectedWholesaleServices", typeof(RejectedWholesaleServices));
-        mapper.Add("AcceptedWholesaleServicesSerieCommand", typeof(AcceptedWholesaleServices));
+        mapper.Add("GridAreaOwnershipAssigned", typeof(GridAreaOwnershipAssignedDto), 0);
+        mapper.Add("SendAggregatedMeasureDataRequestToWholesale", typeof(SendAggregatedMeasureDataRequestToWholesale), 0);
+        mapper.Add("AcceptedAggregatedTimeSerie", typeof(AcceptedEnergyResultTimeSerieCommand), 0);
+        mapper.Add("AcceptedAggregatedTimeSerie", typeof(AcceptedEnergyResultTimeSeriesCommand), 1);
```
Comment (MadsDue) [inline source/Process.Application/Extensions/DependencyInjection/InternalCommandExtensions.cs:49]: If we rename AcceptedAggregatedTimeSerie to AcceptedAggregatedTimeSeries for the new command, would we then need version number?

**Pull request closed without merging**